rust/dyn-ddlog/src/ast/rewrites/block_simplifier.rs

```diff
@@ -0,0 +1,45 @@
+use super::ExpressionRewriter;
+use crate::ast::{Expr, ExprKind};
+use std::mem;
+
+/// Simplifies block statements
+pub struct BlockSimplifier;
+
+impl BlockSimplifier {
+    pub fn new() -> Self {
+        Self
+    }
+}
+
+// TODO: Make sure to collapse spans whenever those are implemented
+impl<I> ExpressionRewriter<I> for BlockSimplifier {
+    /// Collapse redundant blocks, e.g. { 1 } => 1
+    fn rewrite_block(&mut self, mut block: Expr<I>) -> Expr<I> {
+        if let ExprKind::Block(block_exprs) = &mut block.kind {
+            let mut idx = 0;
+            while let Some(expr) = block_exprs.get_mut(idx) {
+                *expr = self.rewrite(mem::take(expr));
```

Comment:
Not until [`.drain_filter()`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.drain_filter) stabilizes, it could be a re-collected iterator I guess but that's kinda wasteful

Reply:
> but that's kinda wasteful

yep, gotta love functional data types.

Reply:
I just started reading about `egg` and do not yet know where you're heading with that, but can't `egg` be used to perform these simplifications?  Would that be an overkill?